test/FunctionalTests/Microsoft.OData.Core.Tests/ScenarioTests/UriParser/ParameterAliasFunctionalTests.cs

```diff
@@ -206,6 +206,71 @@ public void ParsePath_IntArgumentOnShortParameter()
             parseUri.Throws<ODataException>(ODataErrorStrings.MetadataBinder_CannotConvertToType("Edm.Int32", "Edm.Int16"));
         }
 
+        [Fact]
+        public void ParsePath_AliasInUnboundFunction_SingleQuoteWithinDoubleQuotedString()
+        {
+            const string uri = "http://gobbledygook/GetRating(film=@p)?@p={\"@odata.type\":\"%23Fully.Qualified.Namespace.Film\",\"ID\":1,\"Title\":\"The Ogre's Lair\"}";
+
+            ParseUriAndVerify(
+                new Uri(uri),
+                (oDataPath, filterClause, orderByClause, selectExpandClause, aliasNodes) =>
+                {
+                    oDataPath.LastSegment.ShouldBeOperationImportSegment(HardCodedTestModel.GetFunctionImportForGetRating());
+
+                    var constNode = Assert.IsType<ConstantNode>(aliasNodes["@p"]);
+                    Assert.Equal("{\"@odata.type\":\"#Fully.Qualified.Namespace.Film\",\"ID\":1,\"Title\":\"The Ogre's Lair\"}", constNode.Value);
+                });
+        }
+
+        [Fact]
+        public void ParsePath_AliasInUnboundFunction_EscapedDoubleQuotesWithinDoubleQuotedString()
+        {
+            const string uri = "http://gobbledygook/GetRating(film=@p)?@p={\"@odata.type\":\"%23Fully.Qualified.Namespace.Film\",\"ID\":2,\"Title\":\"The \\\"Benevolent\\\" Dictator\"}";
+
+            ParseUriAndVerify(
+                new Uri(uri),
+                (oDataPath, filterClause, orderByClause, selectExpandClause, aliasNodes) =>
+                {
+                    oDataPath.LastSegment.ShouldBeOperationImportSegment(HardCodedTestModel.GetFunctionImportForGetRating());
+
+                    var constNode = Assert.IsType<ConstantNode>(aliasNodes["@p"]);
+                    Assert.Equal("{\"@odata.type\":\"#Fully.Qualified.Namespace.Film\",\"ID\":2,\"Title\":\"The \\\"Benevolent\\\" Dictator\"}", constNode.Value);
+                });
+        }
+
+        [Fact]
+        public void ParsePath_AliasInUnboundFunction_SingleAndEscapedDoubleQuotesWithinDoubleQuotedString()

```

Comment:
Seems to me like these tests could be DRY'd since the logic is the same

Reply:
@odero Fixed